packages/generators/utils/language.ts

```diff
@@ -0,0 +1,117 @@
+import { Rule } from "../types";
+
+export enum LangType {
+	ES6 = "ES6",
+	Typescript = "Typescript",
+}
+
+const replaceExt = (path: string, ext: string): string =>
+	path.substr(0, path.lastIndexOf(".")) + `${ext}'`;
+
+function updateEntryExt(self, newExt: string): void {
+	const jsEntryOption = self.configuration.config.webpackOptions.entry;
+	let tsEntryOption = {};
+	if (typeof jsEntryOption === "string") {
+		tsEntryOption = replaceExt(jsEntryOption, newExt);
+	} else if (typeof jsEntryOption === "object") {
+		Object.keys(jsEntryOption).forEach((entry: string): void => {
+			tsEntryOption[entry] = replaceExt(jsEntryOption[entry], newExt);
+		});
+	}
+	self.configuration.config.webpackOptions.entry = tsEntryOption;
+}
+
+const getFolder = (path: string): string =>
+	path.replace("'./", "").split("/").slice(0, -1).join("/");
+
+function getEntryFolders(self): string[] {
```

Comment:
This splits an entry property to an object if its an object and to a string if it's a string? If so, I think maybe we should name this to `getEntryProperties`

Reply:
It returns an array of strings with the paths of all the folders containing an entry point

For example:
```ts
entry: string = "./this/is/sparta.js"
// getEntryFolders returns ["this/is"]

entry: object = {
  pageOne: './this/is/pageone.js'
  pageTwo: './this/pageTwo.js'
} 
// getEntryFolders returns ["this/is", "this"]
```

I use this inside the babel/typescript loaders configuration to include the folders containing an entry point :)

Reply:
Ok thanks for informing